foolbox/attacks/gradient_descent_base.py

```diff
@@ -65,19 +65,23 @@ def run(
         criterion_ = get_criterion(criterion)
         del inputs, criterion, kwargs
 
-        if not isinstance(criterion_, Misclassification):
+        # perform a gradient ascent (targeted attack) or descent (untargeted attack)
+        if isinstance(criterion_, Misclassification):
+            gradient_step_sign = 1.0
+            classes = criterion_.labels
+        elif isinstance(criterion_, TargetedMisclassification):
```

Comment:
I think it would be nice if this check would be `hasattr` `target_classes` (Python duck typing… allows alternative implementations of a targeted criterion, e.g. with additional confidence checks or so, without enforcing explicit subclassing).

Reply:
Good idea!